src/BaseNode.ts

```diff
@@ -43,11 +47,13 @@ class BaseNode {
     }
   }
 
+  // TODO: none of these destructured values are ever used
+  // should they be here?
```

Comment:
Which values?  begValue, endValue, attr and nameSpace?  I want the user of BaseNode to implement getInterpolator.  Maybe there's a better way to do this in Typescript.

Reply:
I've made both `BaseNode` and `getInterpolator` abstract.  I generally like to avoid inheritance but in this case it seems justified.

Reply:
Yep.  Wasn't crazy about this either but I'm not advertising this to general users.  It's being used in a few libraries internally and people are aware of what's going on.  

Right on.  Yeah, would have taken me awhile to figure this out with the "abstract" part.  Looks good.